### DISCUSSION TITLE
Mirth Results Message

### DISCUSSION THREAD

**Discussion by mwatersonPE** [nextgenhealthcare/connect#5256] [Q&A]
Hello,

I am very new to HL7 and have been assigned the task of figuring out if some configurations are possible.

When Mirth sends results message and recipient receives actual message but fails to accept/process it.

1.Does/Can Mirth get notified of that?
2.Would Mirth resend those failed messages? What's the frequency? How many times would it try?
3.If Mirth fails to send message successfully after ## of attempts, could it update a flag in a database?

Thanks,
Mike W

**Comment by pacmano1** [comment]
Assuming this is HL7:

1) Yes, that is what an HL7 ACK/NACK is for assuming the other side sent one (i.e. a send failure could be a timeout or other problem)
2) It can, that is configurable in the UI.
3) Yes, you have to code for that if you want to record that in your own database.  Mirth of course does have the message status in its own database.

That same concept of course applies to API development also.

**Comment by jonbartels** [comment]
Regarding:

> f Mirth fails to send message successfully after ## of attempts, could it update a flag in a database?

https://appsbyaaron.wordpress.com/2017/11/01/force-a-queued-message-to-error-if-the-number-of-send-attempts-exceeds-a-threshold/ may be informative. This is a common use case where after some number of attempts a message is intentionally failed by the response transformer. Your use case might be able to use similar code to set a flag.

Also note that MC *already* tracks send attempts. Go to a channel in the MC dashboard, open it so you can see that channels messages, right click on the table header, enable the "Send Attempts" column. Since that data is already visible in the dashboard it is inherently stored in the MC database. O
